### PR TITLE
Add Convert.asFrame function

### PR DESCRIPTION
**Pull request by koperagen**

### CODE REVIEW

File: docs/StardustDocs/topics/convert.md [hunk 1]
```diff
@@ -24,6 +25,17 @@ df.convert { colsAtAnyDepth().colsOf<String>() }.with { it.toCharArray().toList(
 <dataFrame src="org.jetbrains.kotlinx.dataframe.samples.api.Modify.convert.html"/>
 <!---END-->
 
+ColumnGroup can be converted using DataFrame API, for example:
+
+<!---FUN convertAsFrame-->
+
+```kotlin
+df.convert { name }.asFrame { it.add("fullName") { "$firstName $lastName" } }
```
Comment (zaleslaw) [inline docs/StardustDocs/topics/convert.md:33]: Cool, doc updates!

File: core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/convert.kt [hunk 2]
```diff
@@ -111,6 +114,9 @@ public inline fun <T, C, reified R> Convert<T, C>.with(
     noinline rowConverter: RowValueExpression<T, C, R>
 ): DataFrame<T> = with(Infer.Nulls, rowConverter)
 
+public fun <T, C, R> Convert<T, DataRow<C>>.asFrame(body: ColumnsContainer<T>.(ColumnGroup<C>) -> DataFrame<R>): DataFrame<T> =
```
Comment (zaleslaw) [inline core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/convert.kt:117]: Is it covered by unit test? please add

**Review by zaleslaw — CHANGES_REQUESTED** [review]
Looks like a JUnit test is missed

File: core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/convert.kt (hunk 2 above)
Comment (koperagen) [reply core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/convert.kt:117]: Added

**Pull request merged**